src/compiler/checker.ts

```diff
@@ -16015,6 +16022,42 @@ namespace ts {
                         }
                     }
                 }
+                else if (target.flags & TypeFlags.Conditional) {
+                    const c = target as ConditionalType;
+                    let skipFalse = false;
+                    let skipTrue = false;

```

Comment:
Just turn these into `const skipXXX = ...` and use ternary expressions. Shorter and easier to follow.